docs/architecture/adr-049-sign-mode-textual-annex1.md

```diff
@@ -0,0 +1,88 @@
+# ADR 049: SIGN_MODE_TEXTUAL - Annex 1 Value Renderers
+
+## Changelog
+
+- Dec 06, 2021: Initial Draft
+
+## Status
+
+Draft
+
+## Abstract
+
+This Annex describes value renderers, which are used for displaying values in a human-friendly way.
+
+## Value Renderers
+
+Value Renderers describe how values of different types should be automatically rendered.
+
+### `number`
+
+- Applies to `sdk.Dec`, `sdk.Int`, and other numeric types (`uint64`, etc.)
+- Formatting with `,`s for every three integral digits
+- Ex:
+  `1000` -> `1,000`
+  `1000000.00` -> `1,000,000.00`
+
+TODO consider `'` as separator?
+
+### `coin`
+
+- Applies to `Coin`
+- Denoms are converted to `display` denoms using `Metadata` (if available)
+- Amounts are converted to `display` denom amounts and rendered as `number`s above
+- One space between the denom and amount
+- In the future, IBC denoms could maybe be converted to DID/IIDs, if we can find a robust way for doing this (ex. `cosmos:hub:atom`)
+- Ex:
+  - `1000000000uatom` -> `1,000 atom`
```

Comment:
should we always uppercase to match what users see generally on exchanges, wallets...?

Reply:
The current configuration for the Hub is:

```js
Meta: {
  description: 'The native staking token of the Cosmos Hub.',
  base: 'uatom',
  display: 'atom',
  name: '',
  symbol: '',
  denomUnits: [
    { denom: 'uatom', exponent: 0, aliases: [Array] },
    { denom: 'matom', exponent: 3, aliases: [Array] },
    { denom: 'atom', exponent: 6, aliases: [] }
  ]
}
```

My understand is the chain should use upper case for the display denom, as I did here in this testing configuration: https://github.com/cosmos/cosmjs/blob/c192fc9b95ef97e4afbf7f5b94f8e8194ae428a6/scripts/simapp44/template/.simapp/config/genesis.json#L438-L456

But feel free to let me know if there are other best practices.